galasa-managers-parent/galasa-managers-core-parent/dev.galasa.textscan.manager/src/main/java/dev/galasa/textscan/spi/TextScannerImpl.java

```diff
@@ -16,88 +23,255 @@
+		Pattern fp = Pattern.compile("\\Q" + failString + "\\E");
+		return scan(inputStream,p,fp,count);
+	}
+
+	@Override
+	public String scanForMatch(String text, Pattern searchPattern, int occurrence)
+			throws MissingTextException, IncorrectOccurancesException {
+
+		if(occurrence<1) {
+			throw new InvalidParameterException("Invalid occurrance");
+		}
+		Matcher m = searchPattern.matcher(text);
+		int found =0;
+		String searchText = "";
+		boolean find = false;
+		while (occurrence > 0) {
+			find=m.find();
+			if (!find&& found ==0) {
+				throw new MissingTextException("Search Pattern '"+searchPattern+"' was not found");
+			}
+			else if(!find) {
+				throw new IncorrectOccurancesException("Was looking for "+occurrence+" instances of '"+searchPattern+"' but found "+ found);
+			}
+			found++;
+			occurrence--;
+			searchText = m.group();
+		}
+		return searchText;
+	}
+
+	@Override
+	public String scanForMatch(String text, String searchString, int occurrence)
+			throws MissingTextException, IncorrectOccurancesException {
+
+		Pattern p = Pattern.compile("\\Q" + searchString + "\\E",Pattern.DOTALL | Pattern.MULTILINE);
+
+		return scanForMatch(text, p, occurrence);
+	}
+
+	@Override
+	public String scanForMatch(ITextScannable scannable, Pattern searchPattern, int occurrence)
+			throws MissingTextException, IncorrectOccurancesException, ManagerException, IOException {
+		if(scannable.isScannableInputStream()) {
+			return scanForMatch(scannable.getScannableInputStream(),searchPattern,occurrence);	
+		}else if(scannable.isScannableString()) {
+			return scanForMatch(scannable.getScannableString(),searchPattern,occurrence);
+		}
+		/** It should never reach this line, it should either be an Input Steam or String **/
+		return null;
+	}
+
+	@Override
+	public String scanForMatch(ITextScannable scannable, String searchString, int occurrence)
+			throws MissingTextException, IncorrectOccurancesException, ManagerException, IOException {
+		if(scannable.isScannableInputStream()) {
+			return scanForMatch(scannable.getScannableInputStream(),searchString,occurrence);	
+		}else if(scannable.isScannableString()) {
+			return scanForMatch(scannable.getScannableString(),searchString,occurrence);
+		}
+		/** It should never reach this line, it should either be an Input Steam or String **/
+		return null;
+	}
+
+	@Override
+	public String scanForMatch(InputStream inputStream, Pattern searchPattern, int occurrence)
+			throws MissingTextException, IncorrectOccurancesException, IOException {
+		BufferedReader  reader = new BufferedReader (new InputStreamReader(inputStream));
+		CircularFifoQueue<String> queue = new CircularFifoQueue<String>(10);
+		String line = "";
+		String toScan = "";
+		String string = "";
+		int found =0;
+		boolean find = false;
+		int lines = 0;
+		int count = 0;
+		if(occurrence<1) {
+			throw new InvalidParameterException("Invalid occurancies number");
+		}
+		while((line =reader.readLine())!=null) {
```

Comment:
same issues here as in the scan method.   you should be able to use the same code for both.